src/plugins/intel_cpu/src/nodes/mha.cpp

```diff
@@ -0,0 +1,1086 @@
+               const std::vector<size_t>& stridesOut, const std::vector<size_t>& stridesIn) {
+    for (int i0 = 0; i0 < dimsOut[0]; i0++) {
+        for (int i1 = 0; i1 < dimsOut[1]; i1++) {
+            pout[i0 * stridesOut[0] + i1 * stridesOut[1]] = pin[i0 * stridesIn[0] + i1 * stridesIn[1]];
+        }
+    }
+}
+
+void MHA::callBrgemm(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>& brgKernel, const void* pin0, const void* pin1, void* pout, void* wsp) {
+    if (ctx.is_with_amx)
+        amx_tile_configure(ctx.palette);
+    if (ctx.is_with_comp) {
+        brgemm_post_ops_data_t post_ops_data;
+        brgemm_kernel_execute_postops(brgKernel.get(), 1, pin0, pin1, nullptr, pout, pout, post_ops_data, wsp);
+    } else {
+        brgemm_kernel_execute(brgKernel.get(), 1, pin0, pin1, nullptr, pout, wsp);
+    }
+}
+
+template<typename data_type, typename acc_type>
+void MHA::mhaImpl(dnnl::stream strm) {
+    const data_type* pTranspose0In0 = reinterpret_cast<const data_type*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const data_type* pTranspose1In0 = reinterpret_cast<const data_type*>(getParentEdgeAt(1)->getMemoryPtr()->GetPtr());
+    const float* pMulIn1 = reinterpret_cast<const float*>(getParentEdgeAt(2)->getMemoryPtr()->GetPtr());
+    const float* pAddIn1 = reinterpret_cast<const float*>(getParentEdgeAt(3)->getMemoryPtr()->GetPtr());
+    const data_type* pTranspose2In0 = reinterpret_cast<const data_type*>(getParentEdgeAt(4)->getMemoryPtr()->GetPtr());
+    data_type* pout = reinterpret_cast<data_type*>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+
+    parallel_for2d(dimsMatMul0Out[0], dimsMatMul0Out[1], [&](size_t i0, size_t i1) {
+        size_t threadNum = parallel_get_thread_num();
+
+        auto pTranspose0In0_aux = pTranspose0In0 + i0 * strTranspose0In0[0] + i1 * strTranspose0In0[2]; // order 0213
+        auto pTranspose1In0_aux = pTranspose1In0 + i0 * strTranspose1In0[0] + i1 * strTranspose1In0[2]; // order 0231
+
+        auto pAddIn1_aux = pAddIn1 + i0 * strAddIn1[0]; // order 0231
+
+        auto bufferMatMul0In1_local = reinterpret_cast<void*>(bufferMatMul0In1.data() + threadNum * bufferMatMul0In1Size);
+        auto bufferMatMul0Out_local = reinterpret_cast<void*>(bufferMatMul0Out.data() + threadNum * bufferMatMul0OutSize);
+        auto bufferMatMul1In1_local = reinterpret_cast<void*>(bufferMatMul1In1.data() + threadNum * bufferMatMul1In1Size);
+        auto bufferMatMul1Out_local = reinterpret_cast<void*>(bufferMatMul1Out.data() + threadNum * bufferMatMul1OutSize);
+
+        auto pTranspose1Out_aux = brgCopyBKernel0 ? reinterpret_cast<data_type*>(bufferMatMul1In1_local)
+                                                    : reinterpret_cast<data_type*>(bufferMatMul0In1_local);
+        auto pTranspose2In0_aux = pTranspose2In0 + i0 * strTranspose2In0[0] + i1 * strTranspose2In0[2]; // order 0213
+
+        reorder2D(pTranspose1In0_aux, reinterpret_cast<data_type*>(pTranspose1Out_aux), {K0, N0}, {N0, 1},
```

Comment:
It seems like Transpose which kernel we already have

Are `K0` and `N0` so small or not? Could we reuse permute kernel, what do you think?

Reply:
This part definitely should be improved in future: in fact we need to unite it with copyB kernel. Anyway that should be considered as separate task.